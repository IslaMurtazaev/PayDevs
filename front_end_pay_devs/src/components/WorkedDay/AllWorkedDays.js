import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import workedDayActions from "../../actions/workedDay";
import WorkedDay from "./WorkedDay";

class WorkedDays extends Component {
  constructor(props) {
    super(props);
    this.props.getAllWorkedDays();
  }

  render() {
    return (
      <div>
        <h3>Your Worked Days</h3>
        <div>
          {this.props.workedDays.map(workedDay => (
            <WorkedDay
              key={workedDay.id}
              workedDay={workedDay}
              projectId={+this.props.match.params.id}
              monthPaymentId={+this.props.match.params.monthPaymentId}
            />
          ))}
        </div>
        <Link to={`${this.props.location.pathname}/create`}>
          Create new Worked day
        </Link>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    workedDays: state.workedDays
  };
};

const mapDispatchersToProps = (dispatch, ownProps) => ({
  getAllWorkedDays: () =>
    dispatch(workedDayActions.getAll(+ownProps.match.params.monthPaymentId))
});

export default connect(
  mapStateToProps,
  mapDispatchersToProps
)(WorkedDays);
