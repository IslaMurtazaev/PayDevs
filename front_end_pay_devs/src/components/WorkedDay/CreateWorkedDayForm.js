import React from "react";
import { connect } from "react-redux";

import FormikWorkedDay from "../../forms/FormikWorkedDay";
import workedDayActions from "../../actions/workedDay";

const CreateWorkedDayForm = props => {
  return (
    <div>
      <h1>Create Worked Day</h1>
      <FormikWorkedDay onSubmit={props.createWorkedDay} />
    </div>
  );
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    createWorkedDay: values =>
      dispatch(
        workedDayActions.create(
          +ownProps.match.params.id,
          +ownProps.match.params.monthPaymentId,
          values
        )
      )
  };
};

export default connect(
  null,
  mapDispatchToProps
)(CreateWorkedDayForm);
