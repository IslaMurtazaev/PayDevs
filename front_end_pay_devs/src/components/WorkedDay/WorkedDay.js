import React from "react";
import { Link } from "react-router-dom";

const WorkedDay = props => {
  const { workedDay, projectId, monthPaymentId, onRemove } = props;

  return (
    <div className="worked">
      <h4 className="date">
        <b>Date:</b> {workedDay.day}
      </h4>
      <h4 className="paid">
        {workedDay.paid || "not "}
        <b>paid</b>
      </h4>
      <div className="button-group-horizontal">
        <Link
          to={`/project/${projectId}/Monthly/${monthPaymentId}/workedDay/${
            workedDay.id
          }/update`}
        >
          <button
            className="updateButton btn btn-warning updateWorkedDay"
            type="button"
          >
            Update
          </button>
        </Link>
        <button
          type="button"
          className="btn btn-danger removeWorkDay"
          onClick={() => onRemove(workedDay.id)}
        >
          Delete
        </button>
      </div>
    </div>
  );
};

export default WorkedDay;
