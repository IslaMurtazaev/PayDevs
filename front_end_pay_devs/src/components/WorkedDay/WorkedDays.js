import React, { Component } from "react";
import { connect } from "react-redux";

import workedDayActions from "../../actions/workedDay";
import WorkedDay from "./WorkedDay";
import CreateWorkedDayForm from "./CreateWorkedDayForm";

class WorkedDays extends Component {
  componentDidMount() {
    this.props.getAllWorkedDays();
  }

  render() {
    return (
      <div>
        {this.props.workedDays.length > 0 && <h3><b>Your Worked Days</b></h3>}
        <div className="worked">
          {this.props.workedDays.map(workedDay => (
            <WorkedDay
              key={workedDay.id}
              workedDay={workedDay}
              projectId={+this.props.match.params.id}
              monthPaymentId={+this.props.match.params.monthPaymentId}
            />
          ))}
        </div>

        <CreateWorkedDayForm projectId={+this.props.match.params.id} monthPaymentId={+this.props.match.params.monthPaymentId} />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    workedDays: state.workedDays
  };
};

const mapDispatchersToProps = (dispatch, ownProps) => ({
  getAllWorkedDays: () =>
    dispatch(workedDayActions.getAll(+ownProps.match.params.monthPaymentId))
});

export default connect(
  mapStateToProps,
  mapDispatchersToProps
)(WorkedDays);
