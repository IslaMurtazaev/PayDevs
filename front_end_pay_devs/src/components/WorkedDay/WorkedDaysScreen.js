import React, { Component } from "react";

import WorkedDay from "./WorkedDay";
import CreateWorkedDayForm from "./CreateWorkedDayForm";

class WorkedDaysScreen extends Component {
  componentWillMount() {
    let { workedDays, getAllWorkedDays } = this.props;
    let monthPaymentId = this.props.match.params.monthPaymentId;

    if (!workedDays.length || workedDays[0].monthPaymentId !== monthPaymentId)
      getAllWorkedDays();
  }

  render() {
    return (
      <div>
        {this.props.workedDays.length > 0 && (
          <h3 className="workedDaysHeader">
            <b>Your Worked Days</b>
          </h3>
        )}
        <div>
          {this.props.workedDays.map(workedDay => (
            <WorkedDay
              key={workedDay.id}
              workedDay={workedDay}
              projectId={+this.props.match.params.id}
              monthPaymentId={+this.props.match.params.monthPaymentId}
              onRemove={this.props.removeWorkedDay}
            />
          ))}
        </div>

        <CreateWorkedDayForm
          projectId={+this.props.match.params.id}
          monthPaymentId={+this.props.match.params.monthPaymentId}
        />
      </div>
    );
  }
}

export default WorkedDaysScreen;
