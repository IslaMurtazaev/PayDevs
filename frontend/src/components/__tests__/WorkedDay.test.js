import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import WorkedDay from "../WorkedDay";

describe("<WorkedDay />", () => {
  let workedDay;
  let component;
  beforeEach(() => {
    workedDay = {
      id: 1,
      day: "2018-08-24",
      paid: true
    };
    component = shallow(<WorkedDay workedDay={workedDay} />);
  });

  it("renders day based on props.workedDay.date", () => {
    expect(component.find(".date").text()).toBe(
      "Date: 2018-08-24"
    );
  });

  it("renders paid based on props.workedDay.paid", () => {
    expect(component.find(".paid").text()).toBe("paid");
    workedDay.paid = false;
    component = shallow(<WorkedDay workedDay={workedDay} />);
    expect(component.find(".paid").text()).toBe("not paid");
  });

  it("renders remove button based on props.workedDay.onRemove", () => {
    let onRemoveSpy = jest.fn();
    component = shallow(
      <WorkedDay workedDay={workedDay} onRemove={onRemoveSpy} />
    );
    component.find(".btn-danger").simulate("click");
    expect(onRemoveSpy).toBeCalledWith(workedDay.id);
  });

  it("matches previous snapshot", () => {
    expect(toJson(shallow(<WorkedDay workedDay={workedDay} />))).toMatchSnapshot();
  });
});
