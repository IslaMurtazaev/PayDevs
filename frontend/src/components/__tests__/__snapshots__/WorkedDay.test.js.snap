// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkedDay /> matches previous snapshot 1`] = `
<div
  className="worked"
>
  <h4
    className="date"
  >
    <b>
      Date:
    </b>
     
    2018-08-24
  </h4>
  <h4
    className="paid"
  >
    <Component />
    <b>
      paid
    </b>
  </h4>
  <div
    className="button-group-horizontal"
  >
    <Link
      replace={false}
      to="/project/undefined/Monthly/undefined/workedDay/1/update"
    >
      <button
        className="updateButton btn btn-warning updateWorkedDay"
        type="button"
      >
        Update
      </button>
    </Link>
    <button
      className="btn btn-danger removeWorkDay"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
